src/pyk/proof/reachability.py

```diff
@@ -168,9 +210,16 @@ def _check_terminal(self, curr_node: KCFG.Node) -> bool:
                 return True
         return False
 
+    def nonzero_depth(self, node: KCFG.Node) -> bool:
+        init = self.proof.kcfg.init[0]
+        ps = self.proof.kcfg.paths_between(init.id, node.id)
+        for p in ps:
+            if len([n for n in p if type(n) is KCFG.Edge]) >= 2:
+                return True
```

Comment:
```suggestion

        def _path_length(_path: Iterable[KCFG.Successor]) -> int:
            _path = list(_path)
            if len(_path) == 0:
                return 0
            if type(_path[0]) is KCFG.Split or type(_path[0]) is KCFG.Cover:
                return _path_length(_path[1:])
            elif type(_path[0]) is KCFG.NDBranch:
                return 1 + _path_length(path[1:])
            elif type(_path[0]) is KCFG.Edge:
                return _path[0].depth + _path_length(path[1:])
            raise ValueError(f'Cannot handle Successor type: {type(path[0])}')

        path_lengths = map(_path_length, self.proof.kcfg.paths_between(init.id, node.id))
        return sorted(path_lengths)[0] > 0
```

Factor out a function like this to calculate Path length (could live in `kcfg.py` file), and use that directly here.